src/local-fs.ts

```diff
@@ -95,36 +97,35 @@ export class LocalFileSystem implements FileSystem {
         this.watcher.close();
     }
 
-    async saveFile(fullPath: string, newContent: string): Promise<Correlation> {
-        const correlation = makeCorrelationId();
-        await this.crud.saveFile(fullPath, newContent);
-        this.registerCorrelator(['directoryCreated'], correlation, e => fullPath.startsWith(e.fullPath), false);
+
+    async saveFile(fullPath: string, newContent: string, correlation:Correlation = makeCorrelationId()): Promise<Correlation> {
+        this.registerCorrelationForPathsInDir(fullPath, correlation, 'directoryCreated')
         this.registerCorrelator(['fileChanged', 'fileCreated'], correlation, e => e.fullPath === fullPath && e.newContent === newContent, true);
+        await this.crud.saveFile(fullPath, newContent);
         return correlation;
     }
 
-    async deleteFile(fullPath: string): Promise<Correlation> {
-        const correlation = makeCorrelationId();
-        await this.crud.deleteFile(fullPath);
+    async deleteFile(fullPath: string, correlation:Correlation = makeCorrelationId()): Promise<Correlation> {
         this.registerCorrelator(['fileDeleted'], correlation, e => e.fullPath === fullPath, true);
+        await this.crud.deleteFile(fullPath);
         return correlation;
     }
 
-    async deleteDirectory(fullPath: string, recursive?: boolean): Promise<Correlation> {
-        const correlation = makeCorrelationId();
-        await this.crud.deleteDirectory(fullPath, recursive);
+    async deleteDirectory(fullPath: string, recursive?: boolean, correlation:Correlation = makeCorrelationId()): Promise<Correlation> {
         this.registerCorrelator(['directoryDeleted'], correlation, e => e.fullPath === fullPath, true);
         if (recursive) {
             const prefix = fullPath + pathSeparator;
             this.registerCorrelator(['directoryDeleted', 'fileDeleted'], correlation, e => e.fullPath.startsWith(prefix), false);
-        }
+        };
+        await this.crud.deleteDirectory(fullPath, recursive);
+
         return correlation;
     }
 
-    async ensureDirectory(fullPath: string): Promise<Correlation> {
-        const correlation = makeCorrelationId();
+    async ensureDirectory(fullPath: string, correlation:Correlation = makeCorrelationId()): Promise<Correlation> {
+        this.registerCorrelationForPathsInDir(fullPath, correlation, 'directoryCreated')
+        // this.registerCorrelator(['directoryCreated'], correlation, e => fullPath.startsWith(e.fullPath), true);
```

Comment:
thanks, any idea why the travis C keeps failing? should i ignore or explore?

Reply:
as far as i see from logs, it can't start headless chrome on linux only, so I would say ignore. 
afaik it is also broken on master right now. Maybe @amir-arad knows

Reply:
sadly it's indeed broken in master. I wo'nt be around to fix it anytime soon, so as long as it fails on loading chrome, it's fine. maybe @AlexShemeshWix can have a go at it?